src/Cache.cpp

```diff
@@ -3222,9 +3318,12 @@ Cache::getAccountData(lmdb::txn &txn, mtx::events::EventType type, const std::st
                 lmdb::val data;
                 if (lmdb::dbi_get(txn, db, lmdb::val(to_string(type)), data)) {
                         mtx::responses::utils::RoomAccountDataEvents events;
-                        mtx::responses::utils::parse_room_account_data_events(
-                          std::string_view(data.data(), data.size()), events);
-                        return events.front();
+                        json j = json::array({
+                          json::parse(std::string_view(data.data(), data.size())),
+                        });
+                        mtx::responses::utils::parse_room_account_data_events(j, events);
+                        if (events.size() == 1)
+                                return events.front();
```

Comment:
What happens when there's more than one event?  Seems like we ignore it.  But shouldn't we still return the most recent one anyway?  Or otherwise somehow handle all of the events?

Reply:
This is actually an ugly hack yo reuse the parser for a list of events to parse a single event. Which is why the event is inserted into a json array at first. One day we should clean that up everywhere in our code :D